packages/odyssey-react-mui/src/labs/Switch.tsx

```diff
@@ -11,42 +11,164 @@
+
+  ...(isDisabled && {
+    p: {
+      color: odysseyDesignTokens.TypographyColorDisabled,
+    },
+    a: {
+      color: `${odysseyDesignTokens.TypographyColorDisabled} !important`,
+    },
+  }),
+}));
+
+const SwitchTrack = styled.div<
+  Pick<SwitchProps, "isChecked" | "isDisabled"> & {
+    odysseyDesignTokens: DesignTokens;
+  }
+>(({ isChecked, isDisabled, odysseyDesignTokens }) => ({
+  position: "relative",
+  width: odysseyDesignTokens.Spacing7,
+  height: `calc(${odysseyDesignTokens.Spacing4} + ${odysseyDesignTokens.Spacing1})`,
+  borderRadius: odysseyDesignTokens.BorderRadiusOuter,
+  backgroundColor: isDisabled
+    ? odysseyDesignTokens.HueNeutral200
+    : isChecked
+      ? odysseyDesignTokens.PaletteSuccessLight
+      : odysseyDesignTokens.HueNeutral300,
+  transition: `background-color ${odysseyDesignTokens.TransitionDurationMain}`,
+}));
+
+const SwitchThumb = styled.span<
+  Pick<SwitchProps, "isChecked" | "isDisabled"> & {
+    odysseyDesignTokens: DesignTokens;
+  }
+>(({ isChecked, isDisabled, odysseyDesignTokens }) => {
+  const thumbOffset = (3 / 16) * (16 / 14);
+  const trackWidth = stripRemFromUnits(odysseyDesignTokens.Spacing7);
+  const thumbWidth =
+    stripRemFromUnits(odysseyDesignTokens.Spacing4) - (2 / 16) * (16 / 14);
+
+  const transformDistance = trackWidth - thumbWidth - thumbOffset * 2;
+
+  return {
+    position: "absolute",
+    top: "50%",
+    left: `${thumbOffset}rem`,
+    width: `calc(${odysseyDesignTokens.Spacing4} - 2px)`,
+    height: `calc(${odysseyDesignTokens.Spacing4} - 2px)`,
+    borderRadius: odysseyDesignTokens.BorderRadiusRound,
+    backgroundColor: isDisabled
+      ? odysseyDesignTokens.HueNeutral50
+      : odysseyDesignTokens.HueNeutralWhite,
+    transform: isChecked
+      ? `translate3d(${transformDistance}rem, -50%, 0)`
+      : "translate3d(0, -50%, 0)",
+    transition: `transform ${odysseyDesignTokens.TransitionDurationMain}`,
+  };
+});
+
+const SwitchCheckMark = styled(CheckIcon)<
+  Pick<SwitchProps, "isChecked" | "isDisabled"> & {
+    odysseyDesignTokens: DesignTokens;
+  }
+>(({ isChecked, isDisabled, odysseyDesignTokens }) => ({
+  position: "absolute",
+  top: "50%",
+  left: 3,
+  width: odysseyDesignTokens.Spacing4,
+  transform: "translateY(-50%)",
+  transition: `opacity ${odysseyDesignTokens.TransitionDurationMain}`,
+  opacity: isChecked ? 1 : 0,
+  path: {
+    fill: isDisabled
+      ? odysseyDesignTokens.HueNeutral50
+      : odysseyDesignTokens.HueNeutralWhite,
+  },
+}));
+
+const HiddenCheckbox = styled.input<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  position: "absolute",
+  top: 0,
+  left: 0,
+  width: "100%",
+  height: "100%",
+  margin: 0,
+  opacity: 0,
+  cursor: "pointer",
+  zIndex: 2,
+
+  "&:focus-visible": {
+    "~ [data-switch-track='true']": {
+      boxShadow: `0 0 0 2px ${odysseyDesignTokens.HueNeutralWhite}, 0 0 0 4px ${odysseyDesignTokens.PalettePrimaryMain}`,
```

Comment:
we have some very specific pixel values here. Are those in our design tokens?

Reply:
We do not. This particular line of code is copy/pasted a lot actually. Anytime we have a focus state, we use this. Not sure if we can pull it into a token though considering it uses other tokens